extensions/vscode-api-tests/src/singlefolder-tests/readonlyFileSystem.test.ts

```diff
@@ -0,0 +1,62 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as assert from 'assert';
+import * as vscode from 'vscode';
+import { TestFS } from '../memfs';
+import { assertNoRpc, closeAllEditors } from '../utils';
+
+suite('vscode API - workspace', () => {
+
+	teardown(async function () {
+		assertNoRpc();
+		await closeAllEditors();
+	});
+
+	test('readonly file system - boolean', async function () {
+		const fs = new TestFS('this-fs', false);
+		const reg = vscode.workspace.registerFileSystemProvider(fs.scheme, fs, { isReadonly: true });
+		// We cannot use the `isWritableFileSystem` API here because it will just return undefined, indicating that we don't know if the file system is writable or not.
+		// assert.strictEqual(vscode.workspace.fs.isWritableFileSystem('this-fs'), false);
```

Comment:
Hm, that is strange though, I would have expected otherwise. I think if we cannot test it that way, we are not really testing the `isReadonly` flag but just the implementation of the `TestFS`. Can we figure out further why `isWriteableFileSystem` would not work?

Reply:
Made the changes discussed offline.